src/Drupal/DrupalAutoloader.php

```diff
@@ -184,7 +184,13 @@ class: Drupal\jsonapi\Routing\JsonApiParamEnhancer
                 tags:
                 - { name: route_enhancer }
                  */
-                unset($serviceDefinition['tags'], $serviceDefinition['calls'], $serviceDefinition['configurator'], $serviceDefinition['factory']);
+                if (is_array($serviceDefinition)) {
+                    foreach (['tags', 'calls', 'configurator', 'factory'] as $key) {
+                        if (isset($serviceDefinition[$key])) {
+                            unset($serviceDefinition[$key]);
+                        }
+                    }
+                }
                 $this->serviceMap[$serviceId] = $serviceDefinition;
```

Comment:
Does the service map break? Do we need to convert the string to an array containing `'class'`?

Reply:
See https://github.com/mglaman/phpstan-drupal/blob/main/src/Drupal/DrupalServiceDefinition.php#L47 

```php
public function __construct(string $id, ?string $class, bool $public = true, ?string $alias = null)
```

I think so. https://github.com/mglaman/phpstan-drupal/blob/main/src/Drupal/ServiceMap.php#L23 this assumes it's an array

```php
    public function setDrupalServices(array $drupalServices): void
    {
        self::$services = [];

        foreach ($drupalServices as $serviceId => $serviceDefinition) {
            if (isset($serviceDefinition['parent'], $drupalServices[$serviceDefinition['parent']])) {
                $serviceDefinition = $this->resolveParentDefinition($serviceDefinition['parent'], $serviceDefinition, $drupalServices);
            }

            // @todo support factories
            if (!isset($serviceDefinition['class'])) {
```